### PR TITLE
refactor: `reached_leader_tick`

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
`reached_leader_tick` has many nested conditional statements which are hard to understand and lack clear comments explaining how this critical function works.

#### Summary of Changes
Unravel the nested conditionals and add comments to improve readability

Fixes #

### CODE REVIEW

**Comment by jstarry** [comment]
@carllin @AshwinSekar can I please get a review on this, thanks!

File: poh/src/poh_recorder.rs [hunk 1]
```diff
@@ -476,26 +476,52 @@ impl PohRecorder {
...
+
+            // If the previous slot was mine, skip grace ticks
+            if self.prev_slot_was_mine(my_pubkey, next_slot) {
+                return true;
+            }
+
+            // If we are not reset to a bank by the previous leader, skip grace
+            // ticks if there isn't a pending reset bank or we aren't configured
+            // to apply grace ticks when we detect a pending reset bank.
+            if !self.is_same_fork_as_previous_leader(next_slot) {
```
Comment (AshwinSekar) [inline poh/src/poh_recorder.rs:506]: Would prefer to remove this nesting as well:
```suggestion
            if self.is_same_fork_as_previous_leader(next_slot) {
              return false;
            }
```

File: poh/src/poh_recorder.rs [hunk 2]
```diff
@@ -476,26 +476,52 @@ impl PohRecorder {
...
-                    // If we are not reset to a bank by the previous leader, skip grace
-                    // ticks unless there is a pending reset bank and we are configured
-                    // to apply grace ticks when we detect a pending reset bank.
-                    !self.is_same_fork_as_previous_leader(next_slot)
-                        && (!self.is_new_reset_bank_pending(next_slot) || {
-                            self.report_pending_fork_was_detected(next_slot);
-                            !self.delay_leader_block_for_pending_fork
-                        })
-                }))
+        if self.tick_height >= ideal_target_tick_height {
```
Comment (AshwinSekar) [inline poh/src/poh_recorder.rs:494]: Can remove nesting here with an early return:
```suggestion
        if self.tick_height < ideal_target_tick_height {
          return false;
        }
```

**Review by AshwinSekar — COMMENTED** [review]
sorry for the delay, just a couple nits

**Comment by jstarry** [comment]
Thanks for reviewing! I'm not strictly opposed to removing the nesting but I think it's more readable to structure the code around the "true" cases so that a reader can see all the nested conditionals required for starting a leader slot. If you read the function wondering in what cases the leader will start their slot, the nesting helps a lot. What do you think?

**Review by AshwinSekar — APPROVED** [review]
makes sense, it's a lot more readable now. thanks!

**Pull request merged**